test/hotspot/jtreg/serviceability/jvmti/ObjectMonitorUsage/ObjectMonitorUsage.java

```diff
@@ -234,35 +251,41 @@ static void test3(boolean isVirtual) throws Error {
         Thread[] eThreads = null;
 
         synchronized (lockCheck) {
-            // entry count: 1
+            // Virtual threads are not supported by the GetObjectMonitorUsage.
+            // Correct the expected values for the virtual thread case.
+            int expEnteringCount = isVirtual ? 0 : NUMBER_OF_ENTERING_THREADS;
+            int expWaitingCount  = isVirtual ? 0 : NUMBER_OF_WAITING_THREADS;
```

Comment:
There are comments below that still reference NUMBER_OF_ENTERING_THREADS  and NUMBER_OF_WAITING_THREADS.

Reply:
Thank you for the comment. In fact, I don't know how to fix it.
Replacing with `expEnteringCount/expWaitingCount` does not make sense to me.
The comments are about the tested pattern, not about the real values.

Reply:
expEnteringCount/expWaitingCount contain the tested patterns. I don't see why they can't just replace NUMBER_OF_ENTERING_THREADS/NUMBER_OF_WAITING_THREADS in the comments also. In fact it is confusing if you don't because code right below the comments references expEnteringCount/expWaitingCount, not NUMBER_OF_ENTERING_THREADS/NUMBER_OF_WAITING_THREADS.

Reply:
...and there are also comments above with this issue.

Reply:
> expEnteringCount/expWaitingCount contain the tested patterns.

I kind of disagree.
Please, take look at the loop below:
```
            for (int i = 0; i < NUMBER_OF_WAITING_THREADS; i++) {
                expEnteringCount = isVirtual ? 0 : NUMBER_OF_ENTERING_THREADS + i + 1;
                expWaitingCount  = isVirtual ? 0 : NUMBER_OF_WAITING_THREADS - i - 1;
                lockCheck.notify(); // notify waiting threads one by one
                // now the notified WaitingTask has to be blocked on the lockCheck re-enter

                // entry count: 1
                // count of threads waiting to enter:       NUMBER_OF_ENTERING_THREADS
                // count of threads waiting to re-enter:    i + 1
                // count of threads waiting to be notified: NUMBER_OF_WAITING_THREADS - i - 1
                check(lockCheck, expOwnerThread(), expEntryCount(),
                      expEnteringCount,
                      expWaitingCount);
            }
```
The comment fixed as you suggest does not look useful anymore as the tested pattern is lost:
```
                // entry count: expOwnerThread()
                // count of threads waiting to enter:       expEnteringCount
                // count of threads waiting to re-enter:   expEntryCount()
                // count of threads waiting to be notified: expWaitingCount
                check(lockCheck, expOwnerThread(), expEntryCount(),
                      expEnteringCount,
                      expWaitingCount);
            }
```

Reply:
I don't understand the issue with the updated commented. It is precisely telling you what the expected "count" values should be. If you leave the macros in the comment, then the comment is wrong for virtual threads. If you want to keep the macros in the comment, you need to add something like "... or 0 for virtual threads".

BTW, the "re-enter" comment should continue to be "i + 1". I'm not sure why it was changed to "expEntryCount()".

Reply:
Okay, please, let me explain this one more time.
The original comments before method `check()` calls describe the testing scenario but not the numbers expected to be returned by the JVMTI `GetObjectMonitorUsage`.
For instance, if the testing scenario says: "count of threads waiting to enter: NUMBER_OF_ENTERING_THREADS" then it means there is a real number of these threads waiting to enter the monitor. And it does not matter if they are platform or virtual threads. They are really waiting to enter the monitor. However, the JVMTI `GetObjectMonitorUsage` won't include virtual threads into the returned results.

Now, I'm suggesting to add the following header for comments before each `check()` method call:
+            // The numbers below describe the testing scenario, not the expected results.
+            // The expected numbers are different for virtual threads because
+            // they are not supported by JVMTI GetObjectMonitorUsage.

Would it work for you?

Reply:
> BTW, the "re-enter" comment should continue to be "i + 1".
> I'm not sure why it was changed to "expEntryCount()".

It depends on what are we trying to describe. We either describe the testing scenario (the number of threads doing something) or the expected results. I understood that you wanted to describe the results instead of the scenario. And then it becomes problematic to do so as you can see.

Reply:
Ok, I understand what you are saying now. I think your clarifying comment above helps a lot.